src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -676,6 +676,27 @@ There's no point in including the leaves of transaction trees.
 void SHAMap::getFetchPack (SHAMap* have, bool includeLeaves, int max,
                            std::function<void (uint256 const&, const Blob&)> func)
 {
+    visitDifferences (have,
+        [includeLeaves, &max, &func] (SHAMapTreeNode& smn) -> bool
+        {
+            if (includeLeaves || smn.isInner ())
+            {
+                Serializer s;
+                smn.addRaw (s, snfPREFIX);
+                func (smn.getNodeHash(), s.peekData());
+
+                if (--max <= 0)
+                    return false;
+            }
+            return true;
+        });
+}
+
+void SHAMap::visitDifferences (SHAMap* have, std::function <bool (SHAMapTreeNode&)> func)
```

Comment:
It would be extra nice to have a quick comment that when `func()` returns `false` then `visitDifferences()` returns to the caller.  I can figure it out by reading the code, but a comment would be quicker to comprehend.